### PR TITLE
Landscaper component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area delivery open-source
/kind enhancement
/priority normal
/platform aws

**What this PR does / why we need it**:
This PR adds the landscaper artifacts for this extension. It consists of a Helm chart that renders the `CloudProfile`, `ControllerRegistration`, `Secrets` and `SecretBinding`s based on the imports and inputs.

/invite @schrodit @Diaphteiros @dguendisch @mandelsoft 

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: .landscaper/deploy_items.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+- name: registration
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:1]:
```suggestion
deployItems:
- name: registration
```

File: .landscaper/deploy_items.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  type: Helm
+  config:
+    apiVersion: helm.deployer.landscaper.gardener.cloud/v1alpha1
+    kind: ProviderConfiguration
+    kubeconfig: {{ .imports.cluster.spec.config.kubeconfig | b64enc }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:6]:
```suggestion
  type: Helm
  target:
    name: {{ .imports.cluster.metadata.name }}
    namespace: {{ .imports.cluster.metadata.namespace }}
  config:
    apiVersion: helm.deployer.landscaper.gardener.cloud/v1alpha1
    kind: ProviderConfiguration
```

`config.kubeconfig` is also possible but specifying the target fits better into the landscaper concepts.

File: .landscaper/deploy_items.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    repository: {{ index .cd.localResources "provider-aws" | .access.image }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:7]:
```suggestion
    repository: {{ index .cd.component.localResources "provider-aws" | .access.image }}
```

File: .landscaper/deploy_items.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    version: {{ index .cd.localResources "provider-aws" | .access.tag }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:8]:
```suggestion
    version: {{ index .cd.component.localResources "provider-aws" | .access.tag }}
```

File: .landscaper/deploy_items.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    name: gardener-extension-provider-aws
+    namespace: garden
+    values:
+      controllerRegistration:
+        deployment:
+          chart: {{ resolve(index .cd.localResources "provider-aws-chart" | .access.imageReference) }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:14]:
```suggestion
          chart: {{ resolve(index .cd.component.localResources "provider-aws-chart" | .access.imageReference) }}
```

File: .landscaper/deploy_items.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+          values:
+            image:
+              repository: {{ index .cd.localResources "provider-aws" | .access.image }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:17]:
```suggestion
              repository: {{ index .cd.component.localResources "provider-aws" | .access.image }}
```

Comment (schrodit) [reply .landscaper/deploy_items.yaml:17]: @ccwienk currently the access for an oci hsa this format:
```
access:
  type: ociRegistry
  imageReference: myimage:1.0.0
```
With that format we do not have direct access to the repoository and the tag.
Should be add that to the component descriptor or should we provide a helper function to parse the tag and repository from the reference?

File: .landscaper/deploy_items.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+              tag: {{ index .cd.localResources "provider-aws" | .access.tag }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:18]:
```suggestion
              tag: {{ index .cd.component.localResources "provider-aws" | .access.tag }}
```

**Review by schrodit — CHANGES_REQUESTED** [review]

File: .landscaper/deploy_items.yaml (hunk 6 above)
Comment (ccwienk) [reply .landscaper/deploy_items.yaml]: @schrodit : As those image-URLs are standardised (from OCI), and widely-used (I think..), and reasonably easy to parse, I would keep it like this, if our "users" prefer it this way.

File: .landscaper/deploy_items.yaml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    apiVersion: helm.deployer.landscaper.gardener.cloud/v1alpha1
+    kind: ProviderConfiguration
+    repository: {{ index .cd.component.localResources "provider-aws" | .access.image }}
+    version: {{ index .cd.component.localResources "provider-aws" | .access.tag }}
+    name: gardener-extension-provider-aws
+    namespace: garden
+    values:
+      controllerRegistration:
+        deployment:
+          chart: {{ resolve(index .cd.component.localResources "provider-aws-chart" | .access.imageReference) }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:17]:
```suggestion
          chart: {{ resolve(index .cd.component.localResources "provider-aws-chart" | .access ) }}
```

File: .landscaper/deploy_items.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+          values:
+            image:
+              repository: {{ index .cd.component.localResources "provider-aws" | .access.image }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:20]:
```suggestion
              repository: {{ ociRefRepo ( index .cd.component.localResources "provider-aws" | .access.imageReference ) }}
```

File: .landscaper/deploy_items.yaml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+              tag: {{ index .cd.component.localResources "provider-aws" | .access.tag }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:21]:
```suggestion
              tag: {{ ociRefVersion  ( index .cd.component.localResources "provider-aws" | .access. imageReference ) }}
```

**Review by schrodit — CHANGES_REQUESTED** [review]

File: .landscaper/deploy_items.yaml [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+- name: registration
+  type: Helm
+  target:
+    name: {{ .imports.cluster.metadata.name }}
+    namespace: {{ .imports.cluster.metadata.namespace }}
+  config:
+    apiVersion: helm.deployer.landscaper.gardener.cloud/v1alpha1
+    kind: ProviderConfiguration
+    repository: {{ index .cd.component.localResources "provider-aws" | .access.image }}
+    version: {{ index .cd.component.localResources "provider-aws" | .access.tag }}
```
Comment (schrodit) [inline .landscaper/deploy_items.yaml:11]:
```suggestion
    chart:
      ref: {{ index .cd.component.localResources "provider-aws" | .access.imageReference }}
```

**Review by schrodit — CHANGES_REQUESTED** [review]

**Review by schrodit — APPROVED** [review]
lgmt from my side.

@Diaphteiros do you see any value that should be configurable for the community?

**Comment by Diaphteiros** [comment]
> lgmt from my side.
> 
> @Diaphteiros do you see any value that should be configurable for the community?

Can't think of something right now.

Looks good.

**Comment by rfranzke** [comment]
Thanks. Can you guys give some quick guidance how to try this out locally? Is there some document for it?

**Comment by schrodit** [comment]
> Thanks. Can you guys give some quick guidance how to try this out locally? Is there some document for it?

currently there is only this documentation how the complete workflow with a component descriptor, a blueprint and remote helm charts can be tested: https://github.com/gardener/landscaper/blob/master/docs/tutorials/01-create-simple-blueprint.md

But this week we are working on a tool to render the deploy items locally.
But I think currently the most effort is the creation of the remote resources (e.g. the helm charts or the component descriptor). 

@ccwienk did you find some time to integrate the remote upload into the ccutils so that all resources are automatically build by the pipeline?

**Review by danielfoehrKn — COMMENTED** [review]
For my understanding: where is the component descriptor?  Should not there be a file called cd.yaml?

**Comment by schrodit** [comment]
> For my understanding: where is the component descriptor? Should not there be a file called cd.yaml?

The component descriptor is generated by the pipeline during the build/release.
Its defined by https://github.com/gardener/gardener-extension-provider-aws/blob/master/.ci/component_descriptor

Currently the component descriptor only contains the oci images but it is planned that all resources like the blueprint and the helm charts are integrated cc @ccwienk

**Comment by rfranzke** [comment]
@schrodit Shall we go ahead and merge this PR as it is now or is some change needed?

**Comment by schrodit** [comment]
I think we can go ahead with this PR

**Pull request merged**